packages/adapter/src/lib/adapter/adapter.js

```diff
@@ -6552,13 +6554,13 @@ class AdapterClass extends EventEmitter {
                 unit: aliasObj.common.unit
             };
 
-            this.aliases[sourceId].targets.push(targetEntry);
+            this.aliases.get(sourceId).targets.push(targetEntry);
 
-            if (!this.aliases[sourceId].source) {
+            if (!this.aliases.get(sourceId).source) {
```

Comment:
IMO we should avoid reading the same item of the map multiple times. With async stuff involved, we could otherwise run into a situation like this (technically this possibility was always there though):
* ensure item exists -> OK
* something else deletes item
* access item -> ERROR

Instead, an access pattern like the following would be better IMO:
* ensure item exists
* read item into local var
* do changes to local var
* if entry is a primitive: write local var back to map

As a nice side effect, converting that to TS is going to be easier since a `Map.get()` always includes `undefined` in the type.

Reply:
Only set it back if it is a primitive? The question is, what if entry has been deleted from set, better always write back?

Reply:
If you edit an object, the variable and the map content refer to the same one. So writing back isn't strictly necessary, but I guess it makes sense for consistency and to avoid async race conditions.

Reply:
Thats why I exactly pointed out the case where the object has been deleted from the set. Because then we will have the case that the temp variable differs from the set, which does no longer contain the object whose reference we have edited. So would always write back, however not sure if this can/should happen, because then we have something like alias sub add and alias sub removage at approx. the same time. We would then need to figure out what is the most recent status.

Reply:
Okay, from my understanding the check at https://github.com/ioBroker/ioBroker.js-controller/pull/1890/files#diff-2e942890ae184c571d716ad3268201e863e6ec5bb71c4e950fe8543a416b0c45R6567 is for such cases, however as we know there are currently related issues #1406

Reply:
In any case, making each operation atomic in the sense that it reads at the beginning and writes at the end should make concurrency issues a bit more manageable I think.